strawberry/arguments.py

```diff
@@ -103,13 +107,16 @@ def convert_argument(
         return value
 
     if isinstance(type_, StrawberryOptional):
-        return convert_argument(value, type_.of_type)
+        return convert_argument(value, type_.of_type, scalar_registry, auto_camel_case)
 
     if isinstance(type_, StrawberryList):
         value_list = cast(Iterable, value)
-        return [convert_argument(x, type_.of_type) for x in value_list]
+        return [
+            convert_argument(x, type_.of_type, scalar_registry, auto_camel_case)
+            for x in value_list
+        ]
 
-    if is_scalar(type_):
+    if is_scalar(type_, scalar_registry):
         return value
```

Comment:
do you think we could remove this check entirely? as we did above? so we don't need to pass the scalar registry?